[Mathspace] Reflow when closing brackets (Fixes #655)

ajhyndman opened the pull request:
Trigger a reflow event when closing a bracketed expression

If this doesn't happen, an edit event won't be triggered when we write an
unmatched expression like `(3, 4]`.  This means that our handler won't
know about the mismatched bracket until we type something else.

Fixes #655

laughinghan commented:
Nice! But, what about the existing cases where `reflow` _is_ triggered, will this result in unnecessary duplicate calls? I notice that `.closeOpposing()` (to which you added a call to `.bubble('reflow')`) is called in `_.createLeftOf()` just before a conditional call to `.bubble('reflow')`, and when deleting a side, which doesn't need to `.bubble('reflow')` because deleting always bubbles reflow anyway in `src/services/keystroke.js`. What about simply moving the `.bubble('reflow')` call in `_.createLeftOf()` out of the conditional, so it just unconditionally does `.bubble('reflow')` when closing a bracket expression by typing?

laughinghan commented:
From Slack:
> Are there any unit tests around event behaviour?
> I found a small bug, and am wondering the best way to demonstrate a proposed fix.

Yep, they're [in `test/publicapi.js`](https://github.com/mathquill/mathquill/blob/master/test/unit/publicapi.test.js#L168-L306), though this should be much simpler than those, something like:
```js
var count = 0;
var mq = MQ.MathField($('<span>').appendTo('#mock')[0], {
  handlers: {
    edit: function() {
      count += 1;
    }
  }
});
mq.typedText('(3, 4');
var countBeforeClosingBracket = count;
mq.typedText(']');
assert.equal(count, countBeforeClosingBracket + 1);
```

ajhyndman commented:
I added a unit test, and tweaked the fix.  Thanks for the tip, I extracted the bubble call from the conditional statement.  That looks cleaner to me..
